mlflow/models/__init__.py

```diff
@@ -81,6 +93,17 @@ def log(cls, artifact_path, flavor, registered_model_name=None, **kwargs):
             mlflow_model = cls(artifact_path=artifact_path, run_id=run_id)
             flavor.save_model(path=local_path, mlflow_model=mlflow_model, **kwargs)
             mlflow.tracking.fluent.log_artifacts(local_path, artifact_path)
+            try:
+                mlflow.tracking.fluent._record_logged_model(mlflow_model)
+            except MlflowException as e:
+                if e.error_code == databricks_pb2.ErrorCode.Name(
+                        databricks_pb2.ENDPOINT_NOT_FOUND):
+                    _logger.warning(
+                        "The current store does not support log model api. "
```

Comment:
```suggestion
                        "MLflow clients 1.7.0 and above attempt to log models to an artifact store and record model "
                        "metadata to the tracking store. The model has been logged to the artifact store under %s, "
                        "but the current tracking store does not support recording model metadata, so model metadata "
                         "will not be logged to the tracking store. If logging to a server via REST, consider "
                         "upgrading the server version to MLflow 1.7.0 or above." % mlflow.get_artifact_uri()) 
```

I think the wording in the last sentence is debatable, but would be good to give users an actionable warning message so they know to upgrade their server if that's the issue (I'm assuming that's the most common case)

Reply:
Yeah makes sense, thanks